apps/chat-e2e/src/tests/promptFolderNumerationTest.test.ts

```diff
@@ -0,0 +1,297 @@
+          ExpectedConstants.newFolderWithIndexTitle(folderNumber),
+        );
+        await promptDropdownMenu.selectMenuOption(MenuOptions.rename);
+        await folderPrompts.editFolderNameWithTick('Renamed Folder');
+
+        await promptBar.createNewFolder();
+        await expect
+          .soft(
+            folderPrompts.getFolderByName(
+              ExpectedConstants.newPromptFolderWithIndexTitle(folderNumber),
+            ),
+            ExpectedMessages.folderIsVisible,
+          )
+          .toBeVisible();
+      },
+    );
+  },
+);
+
+dialTest(
+  'Prompt folder: numeration continues after 999',
+  async ({
+    dialHomePage,
+    promptBar,
+    folderPrompts,
+    promptDropdownMenu,
+    setTestIds,
+  }) => {
+    setTestIds('EPMRTC-2967');
+
+    await dialTest.step('Create a new folder', async () => {
+      await dialHomePage.openHomePage();
+      await dialHomePage.waitForPageLoaded();
+      await promptBar.createNewFolder();
+      await expect
+        .soft(
+          folderPrompts.getFolderByName(
+            ExpectedConstants.newPromptFolderWithIndexTitle(1),
+          ),
+          ExpectedMessages.folderIsVisible,
+        )
+        .toBeVisible();
+    });
+
+    await dialTest.step('Rename the folder to 999', async () => {
+      await folderPrompts.openFolderDropdownMenu(
+        ExpectedConstants.newFolderWithIndexTitle(1),
+      );
+      await promptDropdownMenu.selectMenuOption(MenuOptions.rename);
+      await folderPrompts.editFolderNameWithTick(
+        ExpectedConstants.newPromptFolderWithIndexTitle(999),
+      );
+    });
+
+    await dialTest.step('Create a new folder', async () => {
+      for (let i = 1000; i <= 1002; i++) {
+        await promptBar.createNewFolder();
+        await expect
+          .soft(
+            folderPrompts.getFolderByName(
+              ExpectedConstants.newPromptFolderWithIndexTitle(i),
+            ),
+            ExpectedMessages.folderIsVisible,
+          )
+          .toBeVisible();
+      }
+    });
+  },
+);
+
+dialTest(
+  'Prompt folder: names can be equal on different levels',
+  async ({
+    dialHomePage,
+    promptBar,
+    folderPrompts,
+    promptDropdownMenu,
+    setTestIds,
+  }) => {
+    setTestIds('EPMRTC-2968');
+    const duplicatedFolderName = 'Duplicated Name';
+
+    await dialTest.step('Create four folders', async () => {
+      await dialHomePage.openHomePage();
+      await dialHomePage.waitForPageLoaded();
+      for (let i = 1; i <= 3; i++) {
+        await promptBar.createNewFolder();
+        await expect
+          .soft(
+            folderPrompts.getFolderByName(
+              ExpectedConstants.newPromptFolderWithIndexTitle(i),
+            ),
+            ExpectedMessages.folderIsVisible,
+          )
+          .toBeVisible();
+      }
+    });
+
+    await dialTest.step('Create nested folder hierarchy', async () => {
+      await promptBar.dragAndDropEntityToFolder(
```

Comment:
can be simplified using loop:

`for (let i = 3; i >= 2; i--) {
      await promptBar.dragAndDropEntityToFolder(
        folderPrompts.getFolderByName(
          ExpectedConstants.newPromptFolderWithIndexTitle(i),
        ),
        folderPrompts.getFolderByName(
          ExpectedConstants.newPromptFolderWithIndexTitle(i - 1),
        ),
      );
}`

Reply:
fixed